dear_petition/petition/export/documents/expungable_summary.py

```diff
@@ -62,43 +70,76 @@ def generate_context(batch, contact, petitioner_info):
     """
-
-    offense_records = pm.OffenseRecord.objects.filter(
-        offense__ciprs_record__batch=batch
+    offenses = pm.Offense.objects.filter(
+        ciprs_record__batch=batch
     ).exclude(
-        offense__disposition_method=SUPERSEDING_INDICTMENT_OR_PROCESS
-    ).select_related("offense__ciprs_record__batch")
+        disposition_method=DISP_SUPERSEDING_INDICTMENT
+    ).select_related("ciprs_record__batch")
+
+    return offenses
+
 
+def __create_tables_data(offenses):
+    """
+    Create the data that populates the offense record tables in the document. Organize offense records into a
+    dictionary where the key is a (county, jurisdiction) tuple and the value is a list of offense records for that county
+    and jurisdiction.
+    """
     table_data = {}
 
-    for offense_record in offense_records:
-        county = offense_record.offense.ciprs_record.county
-        jurisdiction = JURISDICTION_MAP[offense_record.offense.ciprs_record.jurisdiction]
+    for offense in offenses:
+
+        county = offense.ciprs_record.county
+        jurisdiction = JURISDICTION_MAP[offense.ciprs_record.jurisdiction]
         key = (county, jurisdiction)
 
-        severity = offense_record.severity
-        offense_date = offense_record.offense.ciprs_record.offense_date
-        offense_record_data = {
-            "file_no": offense_record.file_no,
-            "arrest_date": offense_record.offense.ciprs_record.arrest_date,
-            "description": offense_record.description,
-            "severity": severity[0:1] if severity else None,
-            "offense_date": offense_date.strftime("%Y-%m-%d") if offense_date else None,
-            # if can't get disposition method abbreviation from map, then use disposition method (not the abbreviation)
-            "disposition_method": DISPOSITION_METHOD_CODE_MAP.get(
-                offense_record.offense.disposition_method,
-                offense_record.offense.disposition_method
-            ),
-            "disposed_on": offense_record.disposed_on
-        }
-
-        # append offense record to list for the key, but if key doesn't exist yet, create an empty list first
-        table_data.setdefault(key, []).append(offense_record_data)
+        offense_records = list(offense.offense_records.all())
+        if len(offense_records) not in [1, 2]:
+            logger.error(f"Expecting 1 or 2 offense records, but found {len(offense_records)}")
+
+        guilty_to_lesser = False
+        if offense.verdict == VERDICT_GUILTY and len(offense_records) == 2:
+            same_description = offense_records[0].description == offense_records[1].description
+            same_severity = offense_records[0].severity == offense_records[1].severity
+            if same_description and same_severity:
+                # if offense records same, then remove one
+                offense_records.pop()
```

Comment:
Can you make a helper function like `is_convicted_offense_record()` or similar that encapsulates this logic? If we later decide to modify how we track the convicted offenses, it'd be easier to find in the code and to replace it:

```python
    def is_convicted_offense_record(offense_records):
        if not offense.verdict == VERDICT_GUILTY or not len(offense_records) == 2:
            return False

        same_description = offense_records[0].description == offense_records[1].description
        same_severity = offense_records[0].severity == offense_records[1].severity
        return same_description and same_severity
```

Reply:
Good idea.  I'll change.